### DISCUSSION TITLE
Getting The Context of Generated Instance

### DISCUSSION THREAD

**Discussion by MafSholl** [instancio/instancio#1203] [Q&A] [answered]
I want to ask if it's possible to get the context object of the the currently generated instance. Here is my situation in a piece of code below:

```java
List<Airport> airports = Instancio.ofList(Airport.class)
        .size(5)
        .generate(field(Airport::getCode), gen -> gen.oneOf(destinations))
        .supply(field(Airport::getProvider), gen -> gen.oneOf(savedProviders))
        //the ctx field throws error showing it doesn't exist as far as Instancio's supply() is concerned
        .supply(Select.field(Airport::getCountry), (gen, ctx) -> ctx.get(Select.field(Airport::getCode)))
        .create();
```

My goal for this is: I want to set the value of  `Airport.code` to be same with value of `Airport.country`. Is there any other way to this too- that is, target the value of an already generated field for the same instance and reuse it in another field?

**Comment by armandino** [comment, marked as answer]
You can assign the generated value of one field to another field using the `assign()` method:

```java
List<Airport> airports = Instancio.ofList(Airport.class)
        //...snip
        .assign(Assign.valueOf(field(Airport::getCode)).to(field(Airport::getCountry)))
        .create();
```

Please  see the docs for more examples: https://www.instancio.org/user-guide/#using-assign

**Reply by MafSholl** [reply]
Thanks a lot Armandino, it solved my problem. Although my scenario later change, the assign() method still worked the magic
